src/vm/runners/builtin_runner/range_check.rs

```diff
@@ -73,28 +73,24 @@ impl RangeCheckBuiltinRunner {
         let rule: ValidationRule = ValidationRule(Box::new(
             |memory: &Memory,
              address: &MaybeRelocatable|
-             -> Result<MaybeRelocatable, MemoryError> {
-                match memory.get(address)? {
-                    Some(Cow::Owned(MaybeRelocatable::Int(ref num)))
-                    | Some(Cow::Borrowed(MaybeRelocatable::Int(ref num))) => {
-                        if &BigInt::zero() <= num && num < &BigInt::one().shl(128u8) {
-                            Ok(address.to_owned())
-                        } else {
-                            Err(MemoryError::NumOutOfBounds)
-                        }
+             -> Result<Vec<MaybeRelocatable>, MemoryError> {
+                if let Some(Cow::Borrowed(MaybeRelocatable::Int(ref num))) = memory.get(address)? {
```

Comment:
` Cow::Owned(MaybeRelocatable::Int)` should also be a valid case (Treated as equal as `Cow::Borrowed(MaybeRelocatable::Int)`).
 As we are already calling `to_owned()` on the return value, we can instead call `into_owned() `before the match ( `match memory.get(address)?.into_owned()` and get rid of the Cow before the comparison (to avoid these two cases).